website_sale_require_legal/controllers/main.py

```diff
@@ -25,12 +26,37 @@ def checkout_parse(self, address_type, data, remove_prefix=False,
         return result
 
     def checkout_form_save(self, checkout, *args, **kwargs):
-        """Do not use accepting legal terms to save."""
+        """Do not use accepting legal terms to save and get metadata"""
+        res = super(RequireLegalTermsToCheckout, self).checkout_form_save(
+            checkout, *args, **kwargs)
         if "accepted_legal_terms" in checkout:
             del checkout["accepted_legal_terms"]
-
-        return super(RequireLegalTermsToCheckout, self).checkout_form_save(
-            checkout, *args, **kwargs)
+            user_obj = request.env['res.users']
+            order = request.website.sale_get_order(
+                force_create=1, context=request.context)
+            partner_id = request.env['res.partner']
+            if request.uid != request.website.user_id.id:
+                partner_id = user_obj.browse(request.uid).partner_id
+            elif order.partner_id:
+                user_ids = user_obj.with_context(
+                    dict(request.context, active_test=False)).search(
+                    [("partner_id", "=", order.partner_id.id)])
+                if not user_ids or request.website.user_id.id not in user_ids:
+                    partner_id = order.partner_id
+            if partner_id:
+                environ = request.httprequest.headers.environ
+                metadata = "Website legal terms acceptance metadata: "
+                metadata += "\n".join(
+                    "%s: %s" % (val, environ.get(val)) for val in (
+                            "REMOTE_ADDR",
+                            "HTTP_USER_AGENT",
+                            "HTTP_ACCEPT_LANGUAGE",
+                        )
+                    )
+                website_user = (request.website.salesperson_id.id or
+                                SUPERUSER_ID)
+                partner_id.sudo(website_user).message_post(body=metadata)
```

Comment:
Please explain in README where to find acceptance metadata. One might think it will be in the SO instead.